CONTRIBUTING.md

```diff
@@ -16,120 +16,82 @@ We also use static typing for our function arguments/variables for better code r
+Combine any of the following options to suit your test cases.
 
-### Fixtures
-You can find more information on pytest fixtures [here](https://docs.pytest.org/en/6.2.x/fixture.html).
+- `--local`: Enable local tests.
+- `--s3`: Enable S3 tests.
+- `--hub-cloud`: Enable hub cloud tests.
+- `--memory-skip`: Disable memory tests.
+- `--s3-path`: Specify an s3 path if you don't have access to our internal testing bucket.
+- `--keep-storage`: By default all storages are cleaned up after tests run. Enable this option if you need to check the storage contents.
 
-- `memory_storage`: If `--memory-skip` is provided, tests with this fixture will be skipped. Otherwise, the test will run with only a `MemoryProvider`.
-- `local_storage`: If `--local` is **not** provided, tests with this fixture will be skipped. Otherwise, the test will run with only a `LocalProvider`.
-- `s3_storage`: If `--s3` is **not** provided, tests with this fixture will be skipped. Otherwise, the test will run with only an `S3Provider`.
-- `storage`: All tests that use the `storage` fixture will be parametrized with the enabled `StorageProvider`s (enabled via options defined below). If `--cache-chains` is provided, `storage` may also be a cache chain. Cache chains have the same interface as `StorageProvider`, but instead of just a single provider, it is multiple chained in a sequence, where the last provider in the chain is considered the actual storage.
-- `ds`: The same as the `storage` fixture, but the storages that are parametrized are wrapped with a `Dataset`.
 
-Each `StorageProvider`/`Dataset` that is created for a test via a fixture will automatically have a root created before and destroyed after the test. If you want to keep this data after the test run, you can use the `--keep-storage` option. 
+### Extra Resources
+If you feel lost with any of these sections, try reading up on some of these topics.
 
+- Understand how to write [pytest](https://docs.pytest.org/en/6.2.x/) tests.
+- Understand what a [pytest fixture](https://docs.pytest.org/en/6.2.x/fixture.html) is.
+- Understand what [pytest parametrizations](https://docs.pytest.org/en/6.2.x/parametrize.html) are.
 
-#### Fixture Examples
 
+### Fixture Usage Examples
+These are not all of the available fixtures. You can see all of them [here](hub/tests/).
 
-Single storage provider fixture
+Datasets
 ```python
-def test_memory(memory_storage):
-    # test will skip if `--memory-skip` is provided
-    memory_storage["key"] = b"1234"  # this data will only be stored in memory
-
-def test_local(local_storage):
-    # test will skip if `--local` is not provided
-    memory_storage["key"] = b"1234"  # this data will only be stored locally
-
-def test_local(s3_storage):
-    # test will skip if `--s3` is not provided
-    # test will fail if credentials are not provided
-    memory_storage["key"] = b"1234"  # this data will only be stored in s3
-```
+@enabled_datasets
```

Comment:
is there a good way now for testing a subset of all datasets? For example, I might want to just test local and s3 datasets (and not memory datasets) for transforms.

Reply:
yeah, you just need to write a parametrization, see `enabled_datasets` definition